### PR TITLE
Added Conj Option To net.copy()

### LINKED ISSUE #223: Add conj option to net.copy()
For calculating norms and reduced density matrices, it would be nice to be able to do `net.copy(conj=True)` to conjugate all of the nodes.

We already have conj built into the backends https://github.com/google/TensorNetwork/blob/af16c219e3cc1a411c89314648eb059000c378b8/tensornetwork/backends/base_backend.py#L268 so we can simply use that.

### PR DESCRIPTION
**Pull request by Patil2099**
Fixes #223

### CODE REVIEW

**Comment by Patil2099** [comment]
@Thenerdstation Is this much correct? After this, I will write the Test Cases

File: tensornetwork/network.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def _new_node_name(self, name: Optional[Text]) -> Text:
   def dtype(self) -> Type[np.number]:
     return self.backend.dtype
 
-  def copy(self) -> Tuple["TensorNetwork", dict, dict]:
+  def copy(self,conj=False) -> Tuple["TensorNetwork", dict, dict]:
     """
     Return a copy of the TensorNetwork.
```
Comment (Thenerdstation) [inline tensornetwork/network.py]: Please also add documentation for this argument.

File: tensornetwork/network.py [hunk 2]
```diff
@@ -91,7 +91,7 @@ def copy(self) -> Tuple["TensorNetwork", dict, dict]:
     #TODO: add support for copying CopyTensor
     node_dict = {
         node: new_net.add_node(
-            node.tensor, name=node.name, axis_names=node.axis_names)
+            self.backend.conj(node.tensor), name=node.name, axis_names=node.axis_names)
```
Comment (Thenerdstation) [inline tensornetwork/network.py]: Only conjugate if conj is True

File: tensornetwork/network.py [hunk 3]
```diff
@@ -1231,7 +1231,7 @@ def split_node_full_svd(
     """
     node.reorder_edges(left_edges + right_edges)
     u, s, vh, trun_vals = self.backend.svd_decomposition(
-        node.tensor, len(left_edges), max_singular_values, max_truncation_err)
+        self.backend.conj(node.tensor), len(left_edges), max_singular_values, max_truncation_err)
```
Comment (Thenerdstation) [inline tensornetwork/network.py]: Not sure why you added this line. Please revert

**Comment by Patil2099** [comment]
@googlebot I signed it!

File: tensornetwork/network.py [hunk 4]
```diff
@@ -1231,7 +1239,7 @@ def split_node_full_svd(
     """
     node.reorder_edges(left_edges + right_edges)
     u, s, vh, trun_vals = self.backend.svd_decomposition(
-        node.tensor, len(left_edges), max_singular_values, max_truncation_err)
+      node.tensor, len(left_edges), max_singular_values, max_truncation_err)
```
Comment (Thenerdstation) [inline tensornetwork/network.py]: Revert the spacing

File: tensornetwork/network.py [hunk 5]
```diff
@@ -89,10 +90,17 @@ def copy(self) -> Tuple["TensorNetwork", dict, dict]:
     """
     new_net = TensorNetwork(backend=self.backend.name)
     #TODO: add support for copying CopyTensor
-    node_dict = {
-        node: new_net.add_node(
+    if conj==True:
```
Comment (Thenerdstation) [inline tensornetwork/network.py]: Just do `if conj`

File: tensornetwork/tests/network_test.py [hunk 6]
```diff
@@ -80,8 +80,8 @@ def test_network_copy(backend):
   a[0] ^ b[1]
   a[1] ^ c[2]
   b[2] ^ c[0]
-
-  net_copy, node_dict, _ = net.copy()
+  conj=False
```
Comment (Thenerdstation) [inline tensornetwork/tests/network_test.py]: That's not how passing arguments to a function works.

File: tensornetwork/tests/network_test.py [hunk 7]
```diff
@@ -96,7 +96,8 @@ def test_network_copy_names(backend):
   c = net.add_node(np.random.rand(3, 3, 3), name='c')
   a[0] ^ b[1]
   b[2] ^ c[0]
-  _, node_dict, edge_dict = net.copy()
+  conj=False
```
Comment (Thenerdstation) [inline tensornetwork/tests/network_test.py]: Please do not modify existing tests. Please make a new test and compare the before/after Tensors of a conjugated copy.

File: tensornetwork/network.py (hunk 2 above)
Comment (Patil2099) [reply tensornetwork/network.py]: Can you help me with the Test Cases? @Thenerdstation

**Comment by Thenerdstation** [comment]
I'm going to push directly to your branch to get the tests fixed. I hope thats ok with you!

**Pull request merged**